src/dmd/globals.d

```diff
@@ -478,24 +478,37 @@ alias d_uns64 = uint64_t;
 // file location
 struct Loc
 {
-    const(char)* filename; // either absolute or relative to cwd
+    uint fileIndex;
     uint linnum;
     uint charnum;
```

Comment:
Can we move the line and column numbers to the table as well? Then we wouldn’t need to pass `Loc` by reference.